src/main/java/com/goncharenko/currencyexchangerate/dao/impl/BankRepositoryImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.goncharenko.currencyexchangerate.dao.impl;
+
+import com.goncharenko.currencyexchangerate.dao.BankRepository;
+import com.goncharenko.currencyexchangerate.dao.CurrencyRepository;
+import com.goncharenko.currencyexchangerate.domain.Bank;
+import com.goncharenko.currencyexchangerate.mapper.BankRowMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Optional;
+
+
+@Component
+public class BankRepositoryImpl implements BankRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BankRepositoryImpl.class);
+    private final JdbcTemplate jdbcTemplate;
+    private final CurrencyRepository currencyRepository;
```

Comment:
It is bad practice to use another repository in current repository. Repository is a simple and "dump" class that do a single action, it does not serve as a logic class